test-suite/output/ErrorLocation_12774_1.out

```diff
@@ -1,3 +1,3 @@
-File "./output/ErrorLocation_12774_1.v", line 2, characters 13-14:
+File "./output/ErrorLocation_12774_1.v", line 2, characters 7-14:
```

Comment:
Losing the location precision isn't great. I guess uconstr loses the ast locations? Could that be avoided?

Reply:
No, actually uconstr has access to the locations. Rather the problem is that the heuristic used by Tacinterp to chose the "most precise location" (whatever that means) behaves badly on this example. I personally don't understand that part of the code, but the criterion should be refined to behave correctly on higher-order arguments of tactics, which essentially what `uconstr` is.

Reply:
Where's that code?

Reply:
Stuff around `Tacinterp.catch_error_with_trace_loc` should be a good lead.

Reply:
Note that there is already the same problem with `refine` in master, where the precise error location is never highlighted. This is fairly annoying when writing complex dependent programs that span several screens, but I've managed to survive it so far.